library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -286,149 +146,119 @@ public AbstractJdbcPollInputOperator()
   public void setup(OperatorContext context)
   {
     super.setup(context);
+    intializeDSLContext();
+    if (scanService == null) {
+      scanService = Executors.newScheduledThreadPool(partitionCount);
+    }
     spinMillis = context.getValue(Context.OperatorContext.SPIN_MILLIS);
     execute = true;
     cause = new AtomicReference<Throwable>();
-    emitQueue = new ArrayBlockingQueue<List<T>>(queueCapacity);
-    this.context = context;
+    emitQueue = new LinkedBlockingDeque<>(queueCapacity);
     operatorId = context.getId();
+    windowManager.setup(context);
+  }
 
-    try {
+  private void intializeDSLContext()
+  {
+    create = DSL.using(store.getConnection(), JDBCUtils.dialect(store.getDatabaseUrl()));
+  }
 
-      //If its a range query pass upper and lower bounds
-      //If its a polling query pass only the lower bound
-      if (getRangeQueryPair().getValue() != null) {
-        ps = store.getConnection()
-            .prepareStatement(
-                JdbcMetaDataUtility.buildRangeQuery(getTableName(), getKey(), rangeQueryPair.getKey(),
-                    rangeQueryPair.getValue()),
-                java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+  @Override
+  public void activate(OperatorContext context)
+  {
+    initializePreparedStatement();
+    long largestRecoveryWindow = windowManager.getLargestRecoveryWindow();
+    if (largestRecoveryWindow == Stateless.WINDOW_ID
+        || context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) > largestRecoveryWindow) {
+      scanService.scheduleAtFixedRate(new DBPoller(), 0, pollInterval, TimeUnit.MILLISECONDS);
+    }
+  }
+
+  protected void initializePreparedStatement()
+  {
+    try {
+      // If its a range query pass upper and lower bounds, If its a polling query pass only the lower bound
+      if (isPollerPartition) {
+        ps = store.getConnection().prepareStatement(buildRangeQuery(rangeQueryPair.getKey(), Integer.MAX_VALUE),
```

Comment:
Are you expecting rangeQueryPair form the user? If so, mark it ```@NotNull```. This is not being set anywhere..